File: src/main/java/nl/hanze/week2/ATM/HasAmount.java
package nl.hanze.week2.ATM;

/**
 * Created by Laurens on 9-10-2017.
 */
public class HasAmount implements ATMState {


    @Override
    public void insertCard(ATM atm) {
        System.out.println("Card already present");
    }

    @Override
    public void ejectCard(ATM atm) {
        System.out.println("Paid amount");
        atm.setState(new IdleState());
    }

    @Override
    public void insertPin(ATM atm) {
        System.out.println("Pin already entered");
    }

    @Override
    public void requestAmount(ATM atm) {
        System.out.println("Amount already requested");
    }
}

File: src/main/java/nl/hanze/week2/assignment45/HasAmount.kt
package nl.hanze.week2.assignment45

/**
 * Created by Laurens on 9-10-2017.
 */
class HasAmount : ATMState {


    override fun insertCard(atm: ATM)  = println("Card already present")

    override fun ejectCard(atm: ATM) {
        println("Paid amount")
        atm.state = IdleState()
    }

    override fun insertPin(atm: ATM) =println("Pin already entered")

    override fun requestAmount(atm: ATM) = println("Amount already requested")
}
